modules/tectonic/resources/tectonic.sh

```diff
@@ -1,30 +1,20 @@
-#!/bin/sh
+#!/usr/bin/env bash
 set -e
 
 KUBECONFIG="$1"
 
-# Setup API Authentication
-KUBECTL="oc --config=$KUBECONFIG"
-
-# Setup helper functions
-
 kubectl() {
-	i=0
-
-	echo "Executing kubectl" "$@"
+	(>&2 echo "Executing kubectl $*")
 	while true
 	do
-		i=$((i+1))
-		[ $i -eq 100 ] && echo "kubectl failed, giving up" && exit 1
-
 		set +e
-		out=$($KUBECTL "$@" 2>&1)
+		out=$(oc --config="$KUBECONFIG" "$@" 2>&1)
 		status=$?
 		set -e
 
-		if echo "$out" | grep -q "AlreadyExists"
+		if grep --quiet "AlreadyExists" <<< "$out"
 		then
-			echo "$out, skipping"
+			(>&2 echo "$out, skipping")
```

Comment:
You shouldn't need a subshell here.  Do you remember why you added it?

And while I prefer `echo "..." >&2` (looks like pushing the string into a pipe ;), I've also seen `echo >&2 "..."`.  I haven't seen folks leading with the redirect (or I see it so rarely I'vd forgotten ;).

Reply:
I was worried about the side effect of rewriting that fd, so I put it in a subshell. Let me test this the way you'd written it and follow up.